src/Motor.Extensions.Hosting.RabbitMQ/BasicPropertiesExtensions.cs

```diff
@@ -1,81 +1,89 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text;
 using CloudNative.CloudEvents;
-using Motor.Extensions.Hosting.Abstractions;
+using Motor.Extensions.Hosting.CloudEvents;
 using Motor.Extensions.Hosting.RabbitMQ.Options;
 using RabbitMQ.Client;
 
 namespace Motor.Extensions.Hosting.RabbitMQ
 {
     public static class BasicPropertiesExtensions
     {
-        public static readonly string CloudEventPrefix = "cloudEvents:";
+        public static string CloudEventPrefix => "cloudEvents:";
+
+        private static readonly List<CloudEventAttribute> IgnoredAttributes = new();
+
+        static BasicPropertiesExtensions()
+        {
+            IgnoredAttributes.AddRange(RabbitMQBindingExtension.AllAttributes);
+            IgnoredAttributes.AddRange(RabbitMQPriorityExtension.AllAttributes);
+        }
 
         public static void Update<T>(this IBasicProperties self, MotorCloudEvent<byte[]> cloudEvent,
-            RabbitMQPublisherOptions<T> options, ICloudEventFormatter cloudEventFormatter)
+            RabbitMQPublisherOptions<T> options)
         {
-            var messagePriority = cloudEvent.Extension<RabbitMQPriorityExtension>()?.Priority ??
-                                  options.DefaultPriority;
+            var messagePriority = cloudEvent.GetRabbitMQPriority() ?? options.DefaultPriority;
             if (messagePriority.HasValue)
                 self.Priority = messagePriority.Value;
-            var dictionary = new Dictionary<string, object>();
 
-            foreach (var attr in cloudEvent.GetAttributes())
+            var headers = new Dictionary<string, object>();
+
+            foreach (var (key, value) in cloudEvent.GetPopulatedAttributes())
             {
-                if (string.Equals(attr.Key, CloudEventAttributes.DataAttributeName(cloudEvent.SpecVersion))
-                    || string.Equals(attr.Key,
-                        CloudEventAttributes.DataContentTypeAttributeName(cloudEvent.SpecVersion))
```

Comment:
@farodin91 can you remember the reason to skip this attribute? The three attributes below are skipped because they are specific to RabbitMQ and therefore handled differently. But what is the reason for `DataContentType`?

Reply:
I used the implementation of https://github.com/cloudevents/sdk-csharp/blob/v1.3/src/CloudNative.CloudEvents.Amqp/AmqpCloudEventMessage.cs#L60 and they also dropped this field.

Reply:
So in the [latest version](https://github.com/cloudevents/sdk-csharp/blob/bedad89613e992bc4e905ee85aa2aef79155e1d7/src/CloudNative.CloudEvents.Amqp/AmqpExtensions.cs#L194), they also added a comment, hinting that they use AMQP's native ContentType Property

Reply:
By the way, we did never use the `ContentType` Property of RabbitMQ messages at all. The field was dropped but not used elsewhere in contrast to the AMQP implementation you referenced above.
We should probably fix this soon and when do that, we can apply the same schema to `compressiontype` and use the native `ContentEncoding` property instead.

Reply:
@FlorianHockmann suggested to do that in a separate PR without publishing a release in between

Reply:
https://github.com/GDATASoftwareAG/motornet/issues/259